### DISCUSSION TITLE
Uncaught TypeError: e.matches is not a function

### DISCUSSION THREAD

**Discussion by pablomadariaga** [alpinejs/alpine#1523] [7. Bugs (archived)]
**cdn.min.js:5 Uncaught TypeError: e.matches is not a function**
    at cdn.min.js:5
    at Array.some (<anonymous>)
    at M (cdn.min.js:5)
    at e (cdn.min.js:5)
    at Array.filter (<anonymous>)
    at Object.Kt [as start] (cdn.min.js:5)
    at cdn.min.js:5

**Comment by ryangjchandler** [comment]
@pablomadariaga Could you please provide a code snippet too.

**Reply by mirkin** [reply]
I also get this when upgrading from alpine 2 to 3

**Comment by pablomadariaga** [comment]
@ryangjchandler 

 I just followed the update guide from alpine 2 to 3, but it gave me the above error, in alpine js 2 I had an x-bind: class in the html element, but I was reading the cases so far and I realized that that generated error In alpine 3, I passed it for the bodysuit but there were more errors, so I decided to return to alpine 2.

**This was my code in alpine 2 bypassing the head and meta tags**


`<html :class="{ 'dark': dark }" x-data="data()">`
`<script src="{{ asset('js/alpine.js') }}"></script>`
`<script src="{{ asset('js/init-alpine.js') }}"></script>`
`<body x-ref="body" @resize.window="setMobile" @keydown.escape="closeSideMenu">`


**This is my code in alpine 3, I already have this defined
document.addEventListener ('alpine: initializing', () => {
     Alpine.data (...)
})**

`<script src="{{ asset('js/alpine.js') }}" defer></script>`
`<script src="{{ asset('js/init-alpine.js') }}" defer></script>`
`<body :class="{ 'dark': dark }" x-data="data"  x-ref="body" @resize.window="setMobile" @keydown.escape="closeSideMenu">`

**Comment by SimoTod** [comment]
I believe the issue is the x-data on the html element that is not supported in alpine v3, if you move it to the body, it should work

**Reply by calebporzio** [reply]
Ah, is this something we should add support for @SimoTod?

**Reply by SimoTod** [reply]
@calebporzio I think you can decide if adding support for it or update the documentation to make it clear. I assume quite a few people were using it to set the dark mode even if it was not advised in v2 to have the component on the HTML tag.
I opened a PR to fix it (#1786) but there is a point I'm not too familiar with so you should have a look first.
